source/win-main/file-manager/file-item.vue

```diff
@@ -230,17 +233,24 @@ export default defineComponent({
       }
       return this.obj.children.filter((e: any) => [ 'file', 'code' ].includes(e.type)).length + ' ' + trans('system.files') || 0
     },
-    countWords: function () {
+    countWordsOrCharsOfDirectory: function () {
       if (this.obj.type !== 'directory') {
         return 0
       }
 
-      const wordCount = this.obj.children
-        .filter((file: any) => file.type === 'file')
-        .map((file: any) => file.wordCount)
-        .reduce((prev: number, cur: number) => prev + cur, 0)
-
-      return trans('gui.words', localiseNumber(wordCount))
+      if (this.shouldCountChars) {
+        const charCount = this.obj.children
+          .filter((file: any) => file.type === 'file')
+          .map((file: any) => file.charCount)
```

Comment:
What you could do is run `map((file: any) => (this.shouldCountChars) ? file.charCount : file.wordCount)`. Then you won't have duplicate logic, and `shouldCountChars` will only be re-evaluated once it changes, so it's a simple lookup.

Reply:
Updated PR. Even did a curious quick performance time measurement check, on a 2,000 MD files "real live" workspace, I could not spot even an additional millisecond spent with the "in-loop" check... so: conviced. :-)